### ISSUE TITLE
New Visualization: Response rate to Bugs

### ISSUE THREAD

**Issue by cdolfi** [oss-aspen/8Knot#112]
**Please describe the background and context for this new visualization**
Response/Close rate to issues with the bug/defect tag

**Describe the perspective you'd like the final visual to give**
We would like to show the response rate to an issue that can be assumed to be associated with a bug. 

**Describe the acceptance criteria for the issue and visualization to be complete**
This should be completed in the finalized visualization tool or as a demo testing tools out

**Additional context**

**Comment by cdolfi** [comment]
@sbrennan98 Lets think more of ways to represent this data for the users, we can think of community managers. We want to to give in a few to how their community responds to bugs. Right now the visualization is looking at the time to close for a few issues in the organization, we want to give an overall view of the activity in general. An option could be a survival analysis curve(https://theforeman.org/2019/03/merge-time-review-complexity.html) or maybe looking at it from a staleness perspective to compare how the community is doing over time. Maybe there is an option in there for a pie chart representation. With whatever view it is, definitely should be able to see it from different time intervals 
@JamesKunstle Do you have ideas on this?

**Comment by sbrennan98** [comment]
Thank you for the resources! I think a survival curve would be a good candidate if changing from a bar graph. The response rate metric in addition to the close rate would certainly serve the user well as that's very important information as well. I will work on adding that metric to the visualization.

**Comment by MoralCode** [comment]
Wanted to check in with this to see where things are at. Have we decided on what type of graph would be best for this?

It sounds to me like this is largely just a specialized/filtered view of the general "issue/PR" response rate/response time type graphs that may already exist in 8Knot.

Is this still something that could use a new viz? if so, what would be the best source for an underlying notebook/analysis to get the viz data from?

**Comment by cdolfi** [comment]
Decided? No, the represents where the discussion has paused 

Best? Not a straightforward answer as each has tradeoffs. Some that come to mind:
- Survival Analysis 
- Some type of iteration on staleness 

I could see this as a filtered view of the Issue/PR staleness graphs but no the response rate as that is looking at comments on a pull request rather than how quickly something is closed/PR opened merged/ etc 

I think that the filter view would be a decent starting point. I would probably recommend a more complex solution to this visualization to answer it "well". Some of the attributes to considered:
- Rate to PR linked to be opened/merged 
- rate of closure with by PR merged
- rate of closure with no solution 
- rate of being labled "wont fix' or something

**Comment by MoralCode** [comment]
is this still within the realm of something that would be a good first issue? seems like theres still a lot of planning needed

**Comment by cdolfi** [comment]
Fair point, I think a sub issue could be made for adding a filter to the issue staleness. That portion is probably a good first issue
